Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/booster/DccBoosterRulesValidator.kt

```diff
@@ -1,16 +1,136 @@
 package de.rki.coronawarnapp.covidcertificate.booster
 
+import androidx.annotation.VisibleForTesting
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.node.ObjectNode
+import dagger.Lazy
+import de.rki.coronawarnapp.covidcertificate.common.certificate.CwaCovidCertificate
 import de.rki.coronawarnapp.covidcertificate.common.certificate.DccData
+import de.rki.coronawarnapp.covidcertificate.common.certificate.DccV1
+import de.rki.coronawarnapp.covidcertificate.recovery.core.RecoveryCertificate
+import de.rki.coronawarnapp.covidcertificate.vaccination.core.VaccinationCertificate
+import de.rki.coronawarnapp.covidcertificate.validation.core.business.wrapper.asEvaluatedDccRule
+import de.rki.coronawarnapp.covidcertificate.validation.core.business.wrapper.asExternalRule
+import de.rki.coronawarnapp.covidcertificate.validation.core.business.wrapper.toZonedDateTime
+import de.rki.coronawarnapp.covidcertificate.validation.core.country.DccCountry
+import de.rki.coronawarnapp.covidcertificate.validation.core.rule.DccValidationRule
 import de.rki.coronawarnapp.covidcertificate.validation.core.rule.EvaluatedDccRule
+import de.rki.coronawarnapp.util.serialization.BaseJackson
+import dgca.verifier.app.engine.DefaultCertLogicEngine
+import dgca.verifier.app.engine.UTC_ZONE_ID
+import dgca.verifier.app.engine.data.CertificateType
+import dgca.verifier.app.engine.data.ExternalParameter
+import kotlinx.coroutines.flow.first
+import timber.log.Timber
+import java.time.ZonedDateTime
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
 class DccBoosterRulesValidator @Inject constructor(
-    private val boosterRulesRepository: BoosterRulesRepository
+    private val boosterRulesRepository: BoosterRulesRepository,
+    private val engine: Lazy<DefaultCertLogicEngine>,
+    @BaseJackson private val objectMapper: ObjectMapper,
 ) {
-    suspend fun validateBoosterRules(dccList: List<DccData<*>>): EvaluatedDccRule? {
-        // TODO
-        throw NotImplementedError()
+    @Suppress("BlockingMethodInNonBlockingContext")
+    suspend fun validateBoosterRules(dccList: List<CwaCovidCertificate>): EvaluatedDccRule? {
+        Timber.tag(TAG).d("validateBoosterRules(dccList=%s)", dccList)
+
+        val boosterRules = boosterRulesRepository.rules.first()
```

Comment:
Use the method _updateBoosterNotificationRules_, once my pr ist merged. That will update the rules and return the most up-to-date rules